packaging/linux/run_keybase

```diff
@@ -111,167 +46,123 @@ write_startup_token() {
-
-  run_redirector
+  systemctl --user import-environment KEYBASE_AUTOSTART
 
   # The keybase.gui.service unit has keybase.service and kbfs.service as
   # dependencies, so we don't have to list them here. But including them lets
   # us report an error if they fail to start. Also prefer `restart` to `start`
   # so that we don't race against the service shutting down.
-  systemctl --user restart keybase.service kbfs.service keybase.gui.service
+  # kbfs.service will attempt to start the redirector, but it isn't an error if
+  # it fails to start, which happens if it is disabled.
+  systemctl --user restart keybase.service kbfs
+  [ "$KEYBASE_NO_GUI" == "1" ] || systemctl --user restart keybase.gui
   write_startup_token "systemd"
 }
 
+run_redirector_in_background() {
+  if ! keybase --use-root-config-file config get --direct --assert-false --assert-ok-on-nil disable-root-redirector &> /dev/null; then
+    return 0
+  fi
+  redirector_log="$logdir/keybase.redirector.log"
+  # An older version of post_install.sh could have made a redirector log
+  # here that's owned by root.  If we can't write to it, then just nuke it
+  # and overwrite.
+  if [ -e "$redirector_log" ] && [ ! -w "$redirector_log" ]; then
+    rm -f "$redirector_log"
+  fi
+  echo Mounting the file system...
+  # We need nohup so the redirector doesn't terminate on shell exit,
+  # but it isn't necessary for keybase/kbfs/gui which autofork.
+  nohup keybase-redirector /keybase >> "$redirector_log" 2>&1 &
+}
+
 start_background() {
-  export KEYBASE_RUN_MODE="${KEYBASE_RUN_MODE:-prod}"
-  export KEYBASE_DEBUG=1
+  mkdir -p "$logdir"
+  mkdir -p "$runtime_dir"
 
   echo Launching keybase service...
   # We set the --auto-forked flag here so that updated clients that try to
   # restart this service will know to re-fork it themselves. That's all it does.
-  keybase -d --log-file="$logdir/keybase.service.log" service --auto-forked &>> "$logdir/keybase.start.log" &
-  echo Mounting the file system...
-  run_redirector
-  kbfsfuse -debug -log-to-file "$mountdir" &>> "$logdir/keybase.start.log" &
-  echo Launching Keybase GUI...
-  /opt/keybase/Keybase &>> "$logdir/Keybase.app.log" &
+
+  keybase --debug --use-default-log-file service --auto-forked &>> "$logdir/keybase.start.log" &
+
+  run_redirector_in_background
+
+  echo Starting KBFS...
+  kbfsfuse -debug -log-to-file &>> "$logdir/keybase.start.log" &
+
+  if [ "$KEYBASE_NO_GUI" != "1" ]; then
+      # For system tray icon due to an upstream Electron issue:
+      # https://github.com/electron/electron/issues/10887
+      # Also exported in systemd keybase.gui unit explicitly
+      export XDG_CURRENT_DESKTOP=Unity
+      export KEYBASE_AUTOSTART="$KEYBASE_AUTOSTART"
+
+      echo Launching Keybase GUI...
+      gui_log="$logdir/Keybase.app.log"
+      # Allow distributions to change the location of the gui as long as it's in PATH
+      if gui_path=$(command -v Keybase); then
+          $gui_path &>> "$gui_log" &
```

Comment:
No need to run in a subshell and save the path. If it's available from `command -v Keybase` then it can be called from the path.